components/Modal.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { ReactElement } from 'react';
+import { createPortal } from 'react-dom';
+
+interface Props {
+  visible: boolean;
+  onCancel: () => void;
+}
+
+export default function Modal(
+  props: React.PropsWithChildren<Props>
+): ReactElement {
+  const { visible, onCancel, children } = props;
+
+  React.useEffect(() => {
+    const handleKeyEvent = (event: KeyboardEvent): void => {
+      switch (event.key) {
+        case 'Escape':
+          onCancel();
+          break;
+        default:
+          break;
+      }
```

Comment:
```suggestion
      if (event.key === 'Escape') {
          onCancel();
      }
```

Is there a reason for using a switch statement here?

Reply:
Resolved by [ad1caa0](https://github.com/sandboxnu/searchneu/pull/157/commits/ad1caa04ef27a4c809a43cceca173daad00311b5)